### PR TITLE
Pull request for Marven Wang's bootcamp

### PR DESCRIPTION
**Pull request by demonking-mw**

### CODE REVIEW

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 1]
```diff
@@ -37,12 +37,43 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float) -> Non
...
+    def findpad(
+        self,
+        current_location: "location.Location",
+        landing_pad_locations: "list[location.Location]",
+    ) -> location.Location:
+        """
+        Find the closest landing pad and return its location
+        """
+        closest_index = 0
+        smallest_distance_squared = 5000000
```
Comment (AaronWang04) [inline modules/bootcamp/decision_waypoint_landing_pads.py:67]: float('inf')

File: modules/bootcamp/decision_waypoint_landing_pads.py [hunk 2]
```diff
@@ -37,12 +37,43 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float) -> Non
...
+
+    def findpad(
+        self,
+        current_location: "location.Location",
+        landing_pad_locations: "list[location.Location]",
+    ) -> location.Location:
+        """
+        Find the closest landing pad and return its location
+        """
+        closest_index = 0
```
Comment (AaronWang04) [inline modules/bootcamp/decision_waypoint_landing_pads.py:66]: what if landing pad list is empty? not within the scope of this bootcamp but something to think about

File: modules/bootcamp/decision_simple_waypoint.py [hunk 3]
```diff
@@ -37,12 +37,21 @@ def __init__(self, waypoint: location.Location, acceptance_radius: float) -> Non
         # ↓ BOOTCAMPERS MODIFY BELOW THIS COMMENT ↓
         # ============
 
-        # Add your own
+        self.has_sent_landing_command = False
+        self.mission_completed = False
```
Comment (AaronWang04) [inline modules/bootcamp/decision_simple_waypoint.py:41]: i dont think this is ever used?

**Review by AaronWang04 — APPROVED** [review]

File: modules/bootcamp/decision_simple_waypoint.py [hunk 4]
```diff
@@ -68,7 +77,22 @@ def run(
...
+                    self.waypoint.location_y - report.position.location_y,
+                )
+            elif not self.has_sent_landing_command:
+                # In this case, the drone need to land
+                print("send landing command")
+                command = commands.Command.create_land_command()
+                self.has_sent_landing_command = True
+        elif report.status == drone_status.DroneStatus.MOVING and self.clearance(report):
+            command = commands.Command.create_halt_command()
+            print("HALT COMMAND SENT")
```
[the remark this answers is not in the document]
Comment (demonking-mw) [reply modules/bootcamp/decision_simple_waypoint.py:95]: Sorry, I forgot to delete them.

**Pull request closed without merging**